controller/src/main/java/org/jboss/as/controller/registry/ImmutableManagementResourceRegistration.java

```diff
@@ -252,4 +253,11 @@
      * @return the set of ordered child types, and and empty set if there are none
      */
     Set<String> getOrderedChildTypes();
+
+    /**
+     * Returns capability definition for this resource.
+     *
+     * @return RuntimeCapability if any registered otherwise null
+     */
+    Set<RuntimeCapability<?>> getCapabilities();
```

Comment:
BTW, I'm fine with renaming AbstractCapability too. ManagedCapability? Just Capability? I'm normally not a fan of using 'AbstractXXX' as a return value or param type in widely used APIs.